### PR TITLE
Feat: support infinity for integer ranges

### PR DESCRIPTION
**Pull request by kodumbeats**
Little needs to be added in in order to support use cases for upper bounds and lower bounds, since PHP has the special constant `INF`. However, validation logic fails when using infinity in integer ranges because:
```php
var_dump(gettype(INF));
// string(6) "double"
```

This PR handles the edge case where integer ranges use infinity.

**Testing**
This change includes appropriate unit tests

### CODE REVIEW

**Review by christyjacob4 — APPROVED** [review]
Looks good @kodumbeats  :+1:

**Review by TorstenDittmann — APPROVED** [review]

File: src/Validator/Range.php [hunk 1]
```diff
@@ -128,6 +129,11 @@ public function isValid($value): bool
 
         switch ($this->format) {
             case self::TYPE_INTEGER:
+                // Accept infinity as an integer
+                // Since gettype(INF) === TYPE_FLOAT
+                if ($value === INF || $value === -INF) {
```
Comment (eldadfux) [inline src/Validator/Range.php:134]: This will overwrite the range check, if a value is INF or -INF still doesn't mean his inside the valid range

Comment (kodumbeats) [reply src/Validator/Range.php:134]: Addressed in c42cce8
